yarn-project/pxe/src/note_processor/note_processor.ts

```diff
@@ -105,7 +105,7 @@ export class NoteProcessor {
       const { txLogs } = encryptedL2BlockLogs[blockIndex];
       const blockContext = l2BlockContexts[blockIndex];
       const block = blockContext.block;
-      const dataStartIndexForBlock = block.startNoteHashTreeSnapshot.nextAvailableLeafIndex;
+      const dataEndIndexForBlock = block.header.state.partial.noteHashTree.nextAvailableLeafIndex;
```

Comment:
You are doing this because you only got the new state, but not had the older state anymore in here right?

Reply:
Yes, exactly. We no longer have a note has tree snapshot from the beginning of the block so I needed to use the "end" one.